### ISSUE TITLE
Add metric which explicity shows failed 0-db connections

### ISSUE THREAD

**Issue by LeeSmet** [threefoldtech/0-stor_v2#72]
Right now the metric is slashed from the exported metrics. Keeping the last state and having a separate map with explicit failed connections is better, and allows time series of failures to check if any particular backend has trouble connecting.

**Comment by scottyeager** [comment]
Would be nice to have this.

**Comment by Mik-TF** [comment]
@LeeSmet @iwanbk Can you check this? Thanks!

**Comment by iwanbk** [comment]
Hi @Mik-TF 

Sure, i can check it.
Quick questions:
1. do you have another requirements other than what mentioned in the issue description?

> Keeping the last state and having a separate map with explicit failed connections is better, and allows time series of failures to check if any particular backend has trouble connecting.

2. AFAIK the data will be gone if 0-stor restarted, is it OK?

It won't be gone if the data scrapped by a central prometheus server.
But if there is central prometheus server, we won't need this feature. CMIIW.

**Comment by scottyeager** [comment]
Hi @iwanbk,

The idea here is to expose the zdb connection status as a 0 or 1 value via Prometheus. Currently this information is only available implicitly by the fact that a given zdb is included or not included in the metrics.

When we scrape the data for the purpose of making time series or firing alerts, it's much simpler to work with values in the explicit form. For example, we want to write an alert rule that says, "if any zdb status indicator goes from 1 to 0, fire an alert." The alternative logic of, "if any zdb's metrics suddenly disappear, fire an alert," is much more difficult, if even possible to codify as such a rule typically.

Zstor doesn't need to provide any time series feature itself. That's indeed handled by the Prometheus instance doing the scraping. Zstor just needs to provide an explicit point in time look at which zdbs are reachable and which are not via its Prometheus endpoint.

**Comment by iwanbk** [comment]
OK, i got the point.

I'll check it today.

**Comment by iwanbk** [comment]
We currently have something like this
```console
data_disk_freespace_bytes{address="127.0.0.1:9901",backend_type="meta",namespace="metans"} 173314732032
data_disk_freespace_bytes{address="127.0.0.1:9902",backend_type="data",namespace="dodo"} 173314727936
```

I'm thinking of adding one label there, maybe `status="up"` / `status=1` / `status="down"` / `status=0`
But it might break current rules, which probably fine anyway.

wdyt?

**Comment by scottyeager** [comment]
I think what we need is something like:

```
connection_status: IntGaugeVec
```

where the labels are address, namespace, and type, and the data is the status as 0 or 1.

**Comment by iwanbk** [comment]
OK, LGTM

**Comment by iwanbk** [comment]
Implemented at #138.

There is also the metrics example there @scottyeager @Mik-TF
